drivers/gpio/gpio_max22190.c

```diff
@@ -0,0 +1,651 @@
+			uint8_t wb_val = max22190_reg_read(dev, MAX22190_WB_REG);
+
+			max22190_update_wb_stat(dev, (uint8_t)wb_val);
+		}
+
+		if (data->fault1.reg_bits.max22190_FAULT2) {
+			/* FAULT2 */
+			data->fault2.reg_raw = max22190_reg_read(dev, MAX22190_FAULT2_REG);
+
+			/* FAULT2_EN */
+			data->fault2_en.reg_raw = max22190_reg_read(dev, MAX22190_FAULT2_EN_REG);
+
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFWBS,
+					data->fault2_en.reg_bits.max22190_RFWBSE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFWBO,
+					data->fault2_en.reg_bits.max22190_RFWBOE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFDIS,
+					data->fault2_en.reg_bits.max22190_RFDISE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_RFDIO,
+					data->fault2_en.reg_bits.max22190_RFDIOE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_OTSHDN,
+					data->fault2_en.reg_bits.max22190_OTSHDNE);
+			PRINT_ERR_BIT(data->fault2.reg_bits.max22190_FAULT8CK,
+					data->fault2_en.reg_bits.max22190_FAULT8CKE);
+		}
+	}
+}
+
+static void max22190_state_get(const struct device *dev)
+{
+	const struct max22190_config *config = dev->config;
+
+	if (gpio_pin_get_dt(&config->fault_gpio))
+		max22190_fault_check(dev);
+
+	/* We are reading WB reg because on first byte will be clocked out DI reg
+	 * on second byte we will ge WB value.
+	 */
+	uint8_t wb_val = max22190_reg_read(dev, MAX22190_WB_REG);
+
+	max22190_update_wb_stat(dev, (uint8_t)wb_val);
+}
+
+static int gpio_max22190_config(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	int err = 0;
+
+	if ((flags & (GPIO_INPUT | GPIO_OUTPUT)) == GPIO_DISCONNECTED) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_SINGLE_ENDED) != 0) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0) {
+		return -ENOTSUP;
+	}
+
+	if (flags & GPIO_INT_ENABLE) {
+		return -ENOTSUP;
+	}
+
+	switch (flags & GPIO_DIR_MASK) {
+	case GPIO_INPUT:
+		LOG_INF("Nothing to do, only INPUT supported");
+		break;
+	default:
+		LOG_ERR("On MAX22190 only input option is available!");
+		return -ENOTSUP;
+	}
+
+	return err;
+}
+
+static void max22190_filter_set(const struct device *dev)
+{
+	const struct max22190_config *config = dev->config;
+
+	for (int ch_n = 0 ; ch_n < 8 ; ch_n++) {
+		max22190_reg_write(dev, MAX22190_FILTER_IN_REG(ch_n), config->filter[ch_n].reg_raw);
+	}
+}
+
+static int max22190_fault_set(const struct device *dev)
+{
+	const struct max22190_data *data = dev->data;
+	int ret = 0;
+
+	ret = max22190_reg_write(dev, MAX22190_FAULT1_EN_REG, data->fault1_en.reg_raw);
+	if (ret)
+		goto exit_fault_set;
+
+	ret = max22190_reg_write(dev, MAX22190_FAULT1_REG, data->fault1.reg_raw);
+	if (ret)
+		goto exit_fault_set;
+
+	ret = max22190_reg_write(dev, MAX22190_FAULT2_EN_REG, data->fault2_en.reg_raw);
+	if (ret)
+		goto exit_fault_set;
+
+	ret = max22190_reg_write(dev, MAX22190_FAULT2_REG, data->fault2.reg_raw);
+	if (ret)
+		goto exit_fault_set;
```

Comment:
Add braces to every `if`, `else`, `do`, `while`, `for` and `switch` body, even for single-line code blocks
https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style

Reply:
Done